spec/unit/puppet_x/puppetlabs/scheduled_task/trigger_spec.rb

```diff
@@ -121,6 +121,23 @@
       expect(canonical).to eq(expected)
     end
 
+    [
+      {'schedule' => 'boot'},
+    ].each do |event_manifest|
+
+      describe 'when validating event based triggers' do
+
+        it 'should allow a nil start_time for event triggers' do
+          expect {subject.class.canonicalize_and_validate(event_manifest)}.to_not raise_error(ArgumentError)
+        end
+
+        it 'should not set a default date' do
+          validated = subject.class.canonicalize_and_validate(event_manifest)
+          expect(validated.has_key?('start_date')).to eq(false)
```

Comment:
I think this can use a [predicate matcher](https://relishapp.com/rspec/rspec-expectations/v/2-4/docs/built-in-matchers/predicate-matchers) instead:

```ruby
expect(validated).not_to have_key('start_date')
```

This will produce better output from rspec in a failure

Reply:
Awesome. That does indeed read better.